libminifi/src/Processor.cpp

```diff
@@ -206,6 +207,29 @@ bool Processor::getProperty(std::string name, std::string &value)
 	}
 }
 
+bool Processor::getDynamicProperty(std::string name, std::string &value)
+{
+	if (isRunning())
+		// Because set property only allowed in non running state, we need to obtain lock avoid rack condition
+		_mtx.lock();
+
+	std::map<std::string, Property>::iterator it = _dynamicProperties.find(name);
+	if (it != _dynamicProperties.end())
+	{
+		Property item = it->second;
+		value = item.getValue();
+		if (!isRunning())
+			_mtx.unlock();
+		return true;
+	}
+	else
+	{
+		if (!isRunning())
```

Comment:
I'm new to this, so forgive the question, but why only unlock when it is not running if the above conditional works  only when it is running? This would imply the mutex is locked when you are running. I likely missed something, though.